pkg/services/ngalert/eval/eval.go

```diff
@@ -702,6 +704,14 @@ func evaluateExecutionResult(execResults ExecutionResults, ts time.Time) Results
 			continue
 		}
 
+		// The query service returns instant vectors from prometheus as scalars. We need to handle them accordingly.
+		if isScalarInstantVector(f) {
+			val := f.Fields[1].At(0).(*float64)
```

Comment:
When reviewing, I was worried about the slice access without bounds checking, but see it's done in the `isScalarInstantVector` function, and the type check too. It would be nice to have the checking and indexing+casting done in the function:

```suggestion
		if val, ok := scalarInstantVector(f); ok {
```